### PR TITLE
Implement linear gradients for the svg widget.

### PR DESCRIPTION
**Pull request by derekdreery**
This PR expands svg support in the svg widget to include linear gradients.

### CODE REVIEW

File: druid/src/widget/svg.rs [hunk 1]
```diff
@@ -192,91 +197,160 @@ impl FromStr for SvgData {
...
-    path.apply_affine(offset_matrix * transform_to_affine(p.transform));
-
-    match &p.fill {
-        Some(fill) => {
-            let brush = color_from_usvg(&fill.paint, fill.opacity);
-            ctx.fill(path.clone(), &brush);
-        }
-        None => {}
+    fn linear_gradient_def(&mut self, lg: &usvg::LinearGradient, ctx: &mut PaintCtx) {
+        // Get start and stop of gradient and transform them to image space (TODO check we need to
```
Comment (derekdreery) [inline druid/src/widget/svg.rs:297]: I've experimented and get the results I expect, so this is a "no".

File: druid/src/widget/svg.rs [hunk 2]
```diff
@@ -192,91 +197,160 @@ impl FromStr for SvgData {
...
-            ctx.stroke(path, &brush, stroke.width.value());
+    fn brush_from_usvg(
+        &self,
+        paint: &usvg::Paint,
+        opacity: usvg::Opacity,
+        ctx: &mut PaintCtx,
+    ) -> Rc<piet::Brush> {
+        match paint {
+            usvg::Paint::Color(c) => {
+                // TODO I'm going to assume here that not retaining colors is OK.
```
Comment (derekdreery) [inline druid/src/widget/svg.rs:326]: Are color brushes always cheap to create?

Comment (cmyr) [reply druid/src/widget/svg.rs:326]: This is not a guarantee made by piet, but ideally I think that the caching of brushes, especially color brushes, should be something that is handled at the piet level. Gradient brushes are definitely not necessarily cheap to create.

**Review by cmyr — APPROVED** [review]
I'm happy when you're happy. 😅

**Comment by derekdreery** [comment]
I'm going to merge this so it doesn't bitrot. There are a few comments that could be tidied up in another pass.

**Pull request merged**